test/src/test/java/hudson/security/HMACConfidentialKeyFIPSTest.java

```diff
@@ -0,0 +1,34 @@
+package hudson.security;
+
+
+import static org.junit.Assert.assertThrows;
+
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.jvnet.hudson.test.FlagRule;
+
+import jenkins.security.HMACConfidentialKey;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class HMACConfidentialKeyFIPSTest {
+   @ClassRule
+   // do not use the FIPS140 class here as that initializes the field before we set the property!
+   public static TestRule flagRule = FlagRule.systemProperty("jenkins.security.FIPS140.COMPLIANCE", "true");
+
+    @Test
+    public void testTruncatedMacOnFips() {
+        HMACConfidentialKey key1 = new HMACConfidentialKey("test", 16);
+        IllegalArgumentException  iae = assertThrows(IllegalArgumentException.class, () -> key1.mac("Hello World"));
+        assertEquals("Supplied length can't be less than 32 on FIPS mode", iae .getMessage());
+    }
+
+    @Test
+    public void testCompleteMacOnNonFips() {
+        HMACConfidentialKey key1 = new HMACConfidentialKey("test", 32);
+        String str = key1.mac("Hello World");
+        assertTrue(str, str.matches("[0-9A-Fa-f]{64}"));
```

Comment:
please do not use assertTrue / assertFalse for any condition that is not **directly** a boolean but a computed boolean.
if it fails you do not get told the reason.
Hamcrest has a [matcher](https://hamcrest.org/JavaHamcrest/javadoc/2.2/org/hamcrest/text/MatchesPattern.html#matchesPattern-java.lang.String-) that will match a string against a pattern